iocage_lib/ioc_json.py

```diff
@@ -964,11 +959,14 @@ def check_config(self, conf, default=False):
                 silent=self.silent,
                 exception=ioc_exceptions.CommandNeedsRoot)
 
-        if not default:
-            jail_conf = self.check_jail_config(conf)
-
         conf['CONFIG_VERSION'] = iocage_conf_version
 
+        # New style thin configuration jails won't have this. Only their
+        # defaults will
+        if thickconfig != 'THICK' and not default:
```

Comment:
This will result in incorrect behavior, we want to return early as the code had it as if it's not a THICK configuration type, the default value is actually the one that gets updated, not each thin jail. Thin jails by design only have the changes the user requested and a few other essentials. So what you were likely seeing is the default property changing, and not the jails. So please revert this change.

This code block is used for THICK configuration types, and default configurations. Which all thin jails inherit from.